src/api.js

```diff
@@ -54,10 +56,110 @@ api.declare({
   // Publish message
   await this.publisher.purgeCache({provisionerId, workerType, cacheName});
 
+  try {
+    await this.CacheBuster.create({
+      workerType,
+      provisionerId,
+      cacheName,
+      before: new Date(),
+      expires: taskcluster.fromNow('1 day'),
+    });
+  } catch (err) {
+    if (err.code !== 'EntityAlreadyExists') {
+      throw err;
+    }
+    let cb = await this.CacheBuster.load({
+      workerType,
+      provisionerId,
+      cacheName,
+    });
+
+    await cb.modify(cacheBuster => {
+      cacheBuster.before = new Date();
+      cacheBuster.expires = taskcluster.fromNow('1 day');
+    });
+  }
+
   // Return 204
   res.status(204).send();
 });
 
+api.declare({
+  method:   'get',
+  route:    '/purge-requests',
+  query: {
+    continuationToken: /./,
+    limit: /^[0-9]+$/,
+  },
+  name:     'allPurgeRequests',
+  output:   SCHEMA_PREFIX_CONST + 'purge-cache-request-list.json#',
+  title:    'All Open Purge Requests',
+  description: [
+    'This is useful mostly for administors to view',
+    'the set of open purge requests. It should not',
+    'be used by workers. They should use the purgeRequests',
+    'endpoint that is specific to their workerType and',
+    'provisionerId.',
+  ].join('\n'),
+}, async function(req, res) {
+  let continuation = req.query.continuationToken || null;
+  let limit = parseInt(req.query.limit || 1000, 10);
+  let openRequests = await this.CacheBuster.scan({}, {continuation, limit});
+  return res.reply({
+    cacheHit: false,
+    continuationToken: openRequests.continuation || '',
+    requests: _.map(openRequests.entries, entry => {
+      return {
+        provisionerId: entry.provisionerId,
+        workerType: entry.workerType,
+        cacheName: entry.cacheName,
+        before: entry.before.toJSON(),
+        expires: entry.expires.toJSON(),
+      };
+    }),
+  });
+});
+
+api.declare({
+  method:   'get',
+  route:    '/purge-requests/:provisionerId/:workerType',
+  name:     'purgeRequests',
+  output:   SCHEMA_PREFIX_CONST + 'purge-cache-request-list.json#',
+  title:    'Open Purge Requests for a provisionerId/workerType pair',
+  description: [
+    'List of caches that need to be purged if they are from before',
+    'a certain time. This is safe to be used in automation from',
+    'workers.',
+  ].join('\n'),
+}, async function(req, res) {
+
+  let {provisionerId, workerType} = req.params;
+  let cacheKey = `${provisionerId}/${workerType}`;
+  let cacheHit = false;
+  this.cacheBusterCache[cacheKey] = Promise.resolve(this.cacheBusterCache[cacheKey]).then(async cacheCache => {
+    if (cacheCache && Date.now() - cacheCache.touched < this.cfg.app.cacheTime * 1000) {
+      cacheHit = true;
+      return cacheCache;
+    }
+    return Promise.resolve({reqs: await this.CacheBuster.query({provisionerId, workerType}), touched: Date.now()});
+  });
+  let openRequests = await this.cacheBusterCache[cacheKey].then(cacheCache => {
+    return cacheCache.reqs;
```

Comment:
Isn't this just: `let openRequests = (await this.cacheBusterCache[cacheKey]).reqs;`